vars/deployPlannerSnapshot.groovy

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/groovy
+
+import java.util.regex.Pattern
+
+def call(body) {
+    // evaluate the body block, and collect configuration into the object
+    def config = [:]
+    body.resolveStrategy = Closure.DELEGATE_FIRST
+    body.delegate = config
+    body()
+
+    def yaml
+    def openShiftProject = config.openShiftProject
+    def openShiftTemplate = config.openShiftTemplate
+    def originalImageName = config.originalImageName
+    def newImageName = config.newImageName
+    def deploymentName = config.githubRepo
+    def providerLabel = config.providerLabel ?: 'fabric8'
+    def project = config.githubProject
+
+    def flow = new io.fabric8.Fabric8Commands()
+    def utils = new io.fabric8.Utils()
+
+    openShiftProject = openShiftProject + '-' + utils.getRepoName()
+    container('clients') {
+        if (!flow.isAuthorCollaborator("", project)){
+            currentBuild.result = 'ABORTED'
+            error 'Change author is not a collaborator on the project, aborting build until we support the [test] comment'
+        }
+
+        yaml = flow.getUrlAsString(openShiftTemplate)
```

Comment:
same pattern here, made me think that yaml is getting overwritten in the line below. how about: 

```groovy

def url = flow.getUrlAsString(openshiftTemplate)
def originalImage = "- image: ${...}"
def newImage = "- image: ${...}"
yaml = Pattern.compiler(originalImage).matcher(url).replaceFirst(newImage)
```

Reply:
thanks, fixed.